src/test/javascript/spec/component/markdown-editor/color-picker-command.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { TranslateModule, TranslateService } from '@ngx-translate/core';
+
+import { AceEditorModule } from 'app/shared/markdown-editor/ace-editor/ace-editor.module';
+import { MockComponent } from 'ng-mocks';
+import { FaIconComponent, FaLayersComponent } from '@fortawesome/angular-fontawesome';
+import { MarkdownEditorComponent } from 'app/shared/markdown-editor/markdown-editor.component';
+import { ArtemisMarkdownEditorModule } from 'app/shared/markdown-editor/markdown-editor.module';
+import { ArtemisTestModule } from '../../test.module';
+import { ColorPickerCommand } from 'app/shared/markdown-editor/commands/colorPicker.command';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+
+describe('ColorPickerCommand', () => {
+    let comp: MarkdownEditorComponent;
+    let fixture: ComponentFixture<MarkdownEditorComponent>;
+
+    function testAddColor(hex: string, color: string) {
+        const command = new ColorPickerCommand();
+        comp.colorCommands = [command];
+        fixture.detectChanges();
+        comp.ngAfterViewInit();
+        comp.aceEditorContainer.getEditor().setValue('test');
+
+        command.execute(hex);
+        expect(comp.aceEditorContainer.getEditor().getValue()).toBe('<span class="' + color + '">test</span>');
+    }
+
+    function testRemoveColor(color: string) {
+        const command = new ColorPickerCommand();
+        comp.colorCommands = [command];
+        fixture.detectChanges();
+        comp.ngAfterViewInit();
+        comp.aceEditorContainer.getEditor().setValue('<span class="' + color + '">test</span>');
+
+        command.execute('#ffffff');
+        expect(comp.aceEditorContainer.getEditor().getValue()).toBe('test');
+    }
+
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, TranslateModule.forRoot(), AceEditorModule, ArtemisMarkdownEditorModule],
+            declarations: [MockComponent(FaIconComponent), MockComponent(FaLayersComponent)],
+            providers: [
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(MarkdownEditorComponent);
+                comp = fixture.componentInstance;
+            });
+    });
+
+    it('should add red color on execute', () => {
```

Comment:
Just a suggestion: I think instead of using a lot of test cases calling the same function, you could have a look at using ``it.each``. See https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
Thanks for the suggestion. I will see if I can implement it.